### ISSUE TITLE
C/C++ syntax highlighting inconsistency.

### ISSUE THREAD

**Issue by jrlanglois** [Microsoft/vscode#16281]
- VSCode Version: 1.7.2
- OS Version: 10.11.6 (El Capitan)

### Example problem:

![screen shot 2016-11-30 at 12 17 13](https://cloud.githubusercontent.com/assets/1908886/20763186/fe1b04ec-b6f6-11e6-9839-ae50a28558bb.png)

### Steps to Reproduce:

Effectively, if you copy and paste the following code into your editor, keeping the leading spacing intact.

```
    #define FOO(x)

    FOO(1)

    #undef FOO
```

If you undo the leading spacing, the highlighting works as expected.

This problem only seems to occur if a macro with parameters is in use (i.e. ` FOO(x)`).

In other words, this works fine:
```
    #define FOO

    FOO

    #undef FOO
```

**Comment by jrlanglois** [comment]
I also noticed this one, where `noexcept` doesn't get properly highlighted in all circumstances: 
![screen shot 2016-11-30 at 14 23 34](https://cloud.githubusercontent.com/assets/1908886/20767842/bd8afe2e-b709-11e6-8379-8bb44966b8ac.png)

Can open a new issue altogether if preferred.

**Comment by mjbvz** [comment]
@jrlanglois yes, please open a separate issue since the `noexcept` problem seems unrelated to the first one. Keeping issues more granular helps us better track problems. Thanks.

**Comment by jrlanglois** [comment]
...and done.

**Comment by aeschli** [comment]
Looks like https://github.com/atom/language-c/issues/173
